### PR TITLE
Puzzle information modal front end

### PR DESCRIPTION
**Pull request by kreklb**
## For Pull Requesters/Reviewers:
All of the following should be true before making a pull request and should be inspected by reviewers:

### User Story
- [x] Acceptance criteria are adhered to and completely finished
- [x] All subtasks on Jira are completed

### Unit Testing
- [x] Unit tests are finished and passing
- [x] Code coverage is >80%

### Documentation/Style
- [x] JSDocs are written for methods and components
- [x] Unclear code has been documented
- [x] Style standards are adhered to

## For Reviewers:
The following should be additionally checked by reviewers:

- [x] Code runs locally
- [x] Visually inspect tests; they should not simply pass, but rather they should be doing something and testing acceptance criteria
- [x] See if "jank" code could be rewritten

## Description
Insert here the description of what your PR does.

### CODE REVIEW

**Comment by kreklb** [comment]
I have 1 or 2 changes on this branch for my other user story by accident, the accountCreation function on DataHelper.js and connecting it to AccountCreation.jsx

**Comment by kreklb** [comment]
Also, the only lines I am not testing are within my .catch on my api call, which I am not sure how to force to error out, but I cannot get to 80 because of this

**Comment by PistolRcks** [comment]
Uhhh what about the one test that is failing on `Error: Not implemented: window.alert` at `PuzzleSelectionPage.jsx:29:7`?

**Comment by kreklb** [comment]
> Uhhh what about the one test that is failing on `Error: Not implemented: window.alert` at `PuzzleSelectionPage.jsx:29:7`?

It isn't failing (on my end at least), its just mad that its trying to alert an error.  This is fixed by one line of code and a beforeEach, but I didn't want to touch tests that I didn't change files for in this pr

**Review by connormckellips — APPROVED** [review]

**Comment by PistolRcks** [comment]
Fair enough. Feel free to merge.

**Comment by PistolRcks** [comment]
Whose code is failing??? Honestly that should get fixed before it goes into main anyways so that the rest of everyone else's checks don't fail randomly

**Comment by PistolRcks** [comment]
Ok IDK how to fix this, will ask Caleb lol

File: client/pages/PuzzleSelectionPage/PuzzleSelectionPage.jsx [hunk 1]
```diff
@@ -1,20 +1,40 @@
...
+  const userID = 1;
+  const userIcon = "https://api.dicebear.com/5.x/adventurer/svg?seed=Gracie&scale=130&radius=20&backgroundType=solid,gradientLinear&randomizeIds=true&backgroundColor=c0aede,b6e3f4,d1d4f9,ffdfbf,ffd5dc";
+
+  const[puzzles, setPuzzles] = useState([{puzzle_id:1, title:"title", description:"description"}]);
+  const[hasResponded, setHasResponded] = useState(true);
+
+  //If puzzles is still default, this evaluates to true
+  //hasResponded makes sure that if there is 1 puzzle in the DB that we don't accidentally
+  //infinitely call the api for puzzles (ran into this issue during testing)
+  if(puzzles.length == 1 && hasResponded) {
```
Comment (Blackkirby72) [inline client/pages/PuzzleSelectionPage/PuzzleSelectionPage.jsx]: `!hasResponded` and start hasResponded as `false`

File: client/pages/PuzzleSelectionPage/PuzzleSelectionPage.jsx [hunk 2]
```diff
@@ -1,20 +1,40 @@
...
+    listPuzzles()
+    .then((res) => {
+      setPuzzles(res.data);
+      setHasResponded(false);
+    })
+    .catch((err) => {
+      alert(err);
```
Comment (Blackkirby72) [inline client/pages/PuzzleSelectionPage/PuzzleSelectionPage.jsx:29]: it's mad about this, but i cant try and look into it

File: client/pages/PuzzleSelectionPage/PuzzleSelectionPage.jsx [hunk 3]
```diff
@@ -1,20 +1,40 @@
...
+  const[hasResponded, setHasResponded] = useState(true);
+
+  //If puzzles is still default, this evaluates to true
+  //hasResponded makes sure that if there is 1 puzzle in the DB that we don't accidentally
+  //infinitely call the api for puzzles (ran into this issue during testing)
+  if(puzzles.length == 1 && hasResponded) {
+    listPuzzles()
+    .then((res) => {
+      setPuzzles(res.data);
+      setHasResponded(false);
```
Comment (Blackkirby72) [inline client/pages/PuzzleSelectionPage/PuzzleSelectionPage.jsx]: `setHasResponded(true)`

**Pull request merged**